gammapy/modeling/fit.py

```diff
@@ -359,23 +359,63 @@ def stat_profile(
 
         return {"values": values, "stat": np.array(stats)}
 
-    def stat_contour(self):
+    def stat_surface(
+        self, x, y, x_values, y_values, reoptimize=False, **optimize_opts
+    ):
         """Compute fit statistic contour.
 
         The method used is to vary two parameters, keeping all others fixed.
         So this is taking a "slice" or "scan" of the fit statistic.
 
+        Caveat: This method can be very computationally intensive and slow. In future releases, parallel computation
+        will make it more efficient.
+
         See also: `Fit.minos_contour`
 
         Parameters
         ----------
-        TODO
+        x, y : `~gammapy.modeling.Parameter`
```

Comment:
There is confusion here. Should the method take `Parameter` directly or parameter names in the form of `str`.
How does it work when two parameters have the  same name?

Reply:
We have to be consistent between `stat_profile`, `stat_surface` and `minos_contour`. Is it OK if we clarify this in a further PR?

Reply:
I'd be fine with this. I think the only way to solve this to access parameters via the model itself, e.g. by introducing something like `Model.get_parameter_by_unique_name("model_name.spectral.parameter_name")`